pkg/virt-controller/services/template.go

```diff
@@ -540,6 +540,11 @@ func (t *templateService) renderLaunchManifest(vmi *v1.VirtualMachineInstance, i
 					})
 			}
 		}
+		sidecarContainer.Env = []k8sv1.EnvVar{
+			k8sv1.EnvVar{
+				Name:  hooks.ContainerNameEnvVar,
+				Value: sidecarContainerName(i),
+			}}
```

Comment:
Please consider moving this logic to `pkg/virt-controller/services/rendercontainer.go` as it already contains similar logic.

Please see `newSidecarContainerRenderer()` above.

Additionally, could it be covered by unit tests?

Reply:
done + added it to ut